### PR TITLE
Build and CI integration for the Provisioning SDK

### PR DESCRIPTION
**Pull request by CIPop**
Adding global build script allowing build/test on Windows/Linux.
Enabling "treat warnings as errors" in all netstandard projects except ServiceClient
Fixing warnings from all netstandard projects except 
Porting DeviceClient Netstandard to the .NET Core SDK toolset.
Updating UWP project.lock.json for VS2017.
Updating WIP Jenkins script. Add DPS sample to the Jenkins build.
Renaming TransportHandler to TransportClient (this also avoids certain internal naming conflicts).

### CODE REVIEW

File: security/dice/src/Microsoft.Azure.Devices.Provisioning.Security.Dice.csproj [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
```
Comment (simonporter) [inline security/dice/src/Microsoft.Azure.Devices.Provisioning.Security.Dice.csproj]: Is Dice the right name here, x509 is what needs to be supported first and foremost, Dice is an extension to that

File: device/Microsoft.Azure.Devices.Client/TransientFaultHandling/AsyncExecution[T].cs [hunk 2]
```diff
@@ -123,7 +123,11 @@ private Task<TResult> ExecuteAsyncContinueWith(Task<TResult> runningTask)
             }
             TimeSpan zero = TimeSpan.Zero;
             Exception innerException = runningTask.Exception.InnerException;
+
+// 'RetryLimitExceededException' is obsolete: 'You should use cancellation tokens or other means of stoping the retry loop.'
```
Comment (simonporter) [inline device/Microsoft.Azure.Devices.Client/TransientFaultHandling/AsyncExecution[T].cs]: nit: spelling, 'stopping'

Comment (simonporter) [reply device/Microsoft.Azure.Devices.Client/TransientFaultHandling/AsyncExecution[T].cs]: other places as well

File: security/dice/src/Microsoft.Azure.Devices.Provisioning.Security.Dice.csproj (hunk 1 above)
Comment (CIPop) [reply security/dice/src/Microsoft.Azure.Devices.Provisioning.Security.Dice.csproj]: Naming is still TBD. This library will support the DICE protocol only.
To support X509, an application must inherit from `SecurityClientX509Certificate` and implement the `GetAuthenticationCertificate()` method. This API is currently placed in the Shared assembly.

**Comment by jasmineymlo** [comment]
> {

check if lock.json is in the .ignore...

---
Refers to: service/Microsoft.Azure.Devices.NetStandard/project.lock.json:1 in c0968ec. [](commit_id = c0968ec63954662a4410d4287abb0a68fb5e7882, deletion_comment = False)

**Comment by CIPop** [comment]
> check if lock.json is in the .ignore...

Thanks! Need to also remove all lock.json files.

File: device/Microsoft.Azure.Devices.Client.NetStandard/Microsoft.Azure.Devices.Client.NetStandard.csproj [hunk 3]
```diff
@@ -189,31 +198,12 @@
     <Compile Include="System.Web\Util\UriUtil.cs" />
     <Compile Include="System.Web\Util\Utf16StringValidator.cs" />
   </ItemGroup>
-  <ItemGroup>
```
Comment (CIPop) [inline device/Microsoft.Azure.Devices.Client.NetStandard/Microsoft.Azure.Devices.Client.NetStandard.csproj]: @jasmineymlo pointed out that we still need the loc generation for now.
Adding this back.

File: device/Microsoft.Azure.Devices.Client/TransientFaultHandling/RetryLimitExceededException.cs [hunk 4]
```diff
@@ -27,7 +27,7 @@ namespace Microsoft.Azure.Devices.Client.TransientFaultHandling
     /// The special type of exception that provides managed exit from a retry loop. The user code can use this
     /// exception to notify the retry policy that no further retry attempts are required.
     /// </summary>
-    [Obsolete("You should use cancellation tokens or other means of stoping the retry loop.")]
+    [Obsolete("You should use cancellation tokens or other means of stopping the retry loop.")]
```
Comment (CIPop) [inline device/Microsoft.Azure.Devices.Client/TransientFaultHandling/RetryLimitExceededException.cs]: Test comment. <please ignore>

**Pull request merged**